build/v1/types.go

```diff
@@ -35,7 +35,7 @@ type BuildSpec struct {
 
 	// triggeredBy describes which triggers started the most recent update to the
 	// build configuration and contains information about those triggers.
-	TriggeredBy []BuildTriggerCause `json:"triggeredBy" protobuf:"bytes,2,rep,name=triggeredBy"`
+	TriggeredBy []BuildTriggerCause `json:"triggeredBy,omitempty" protobuf:"bytes,2,rep,name=triggeredBy"`
```

Comment:
I need to check whether we look at nil vs empty for this one.  @adambkaplan do you know whether there is any controller logic that looks to fill this based on empty vs nil?

Reply:
thanks